OEL-2598: Add optional categories to news items.

AaronGilMartinez opened the pull request.

AaronGilMartinez committed:
OEL-2598: Add install config and override templates.

AaronGilMartinez committed:
OEL-2598: Add update config and hook update.

AaronGilMartinez committed:
OEL-2598: Add field to banner, editor permissions and test.

AaronGilMartinez committed:
OEL-2598: Fix comments.

brummbar commented on tests/src/ExistingSite/NewsTest.php:
```diff
@@ -30,6 +35,45 @@ protected function setUp(): void {
+    $page = $this->getSession()->getPage();
+
+    // Assert user without permission can't create news types.
+    $this->assertPathsRequireRole([
```
Please check how this has been done in https://github.com/openeuropa/oe_showcase/pull/202.

AaronGilMartinez replied:
Moved

brummbar commented on tests/src/ExistingSite/NewsTest.php:
```diff
@@ -60,6 +105,13 @@ public function testCreateNews() {
     ]);
     $media_image->save();
 
+    for ($i = 1; $i < 4; $i++) {
+      Term::create([
+        'vid' => 'news_type',
+        'name' => "News type $i",
+      ])->save();
+    }
```
Please follow https://github.com/openeuropa/oe_showcase/pull/202/files#diff-3b4fc7927c226f71dbc0f511f9c6c951714e1b0f1109cc563db388445b65191aR33 .

AaronGilMartinez replied:
Done

brummbar reviewed (changes requested):
Missing list pages integration.
Fields should be put inside the oe_showcase_fields module.

AaronGilMartinez committed:
OEL-2598: Move taxonomy access to right test file and change terms creation.

AaronGilMartinez committed:
OEL-2598: Move new field to oe_showcase_fields module.

AaronGilMartinez committed:
OEL-2598: Add list pages news type filter.

AaronGilMartinez commented:
> Missing list pages integration. Fields should be put inside the oe_showcase_fields module.

Added filter for news list page and moved field to module

AaronGilMartinez committed:
OEL-2598: Remove unneeded trait.

brummbar commented on modules/oe_showcase_fields/oe_showcase_fields.post_update.php:
```diff
@@ -0,0 +1,21 @@
+use Drupal\oe_bootstrap_theme\ConfigImporter;
+
+/**
+ * Add News type to News.
```
:D

brummbar reviewed (changes requested):
Just this small remark :D

brummbar committed:
OEL-2598: Fix comment.
